### PR TITLE
vulkaninfo: Add surface present modes

### PR DESCRIPTION
**Pull request by joey-lunarg**
Change-Id: Icb6c30a81743cf5854fd7ecc974d87e814be856d

### CODE REVIEW

File: demos/vulkaninfo.c [hunk 1]
```diff
@@ -426,6 +426,21 @@ static const char *vk_format_string(VkFormat fmt) {
     }
 }
 
+static const char *vk_present_mode_string(VkPresentModeKHR mode) {
+    switch (mode) {
+#define STR(r)                                                                 \
```
Comment (mark-lunarg) [inline demos/vulkaninfo.c]: @joey-lunarg, avoid using preprocessor macros if at all possible.

File: demos/vulkaninfo.c [hunk 2]
```diff
@@ -955,15 +970,34 @@ static int app_dump_surface_formats(struct app_instance *inst, struct app_gpu *g
...
     for (uint32_t i = 0; i < formatCount; i++) {
         printf("\t%s\n", vk_format_string(surfFormats[i].format));
     }
-    printf("\n");
     fflush(stdout);
     return formatCount;
 }
+
+static int app_dump_surface_present_modes(struct app_instance *inst, struct app_gpu *gpu) {
+    // Get the list of VkPresentMode's that are supported:
```
Comment (mark-lunarg) [inline demos/vulkaninfo.c]: kill '

File: demos/vulkaninfo.c [hunk 3]
```diff
@@ -955,15 +970,34 @@ static int app_dump_surface_formats(struct app_instance *inst, struct app_gpu *g
...
+    VkResult U_ASSERT_ONLY err;
+    uint32_t presentModeCount = 0;
```
Comment (mark-lunarg) [inline demos/vulkaninfo.c]: vars are lower_camel_case

File: demos/vulkaninfo.c [hunk 4]
```diff
@@ -955,15 +970,34 @@ static int app_dump_surface_formats(struct app_instance *inst, struct app_gpu *g
...
 
     for (uint32_t i = 0; i < formatCount; i++) {
         printf("\t%s\n", vk_format_string(surfFormats[i].format));
     }
-    printf("\n");
     fflush(stdout);
     return formatCount;
 }
+
+static int app_dump_surface_present_modes(struct app_instance *inst, struct app_gpu *gpu) {
```
Comment (mark-lunarg) [inline demos/vulkaninfo.c]: Function names should be UpperCamelCase.  I know it doesn't match, but we've been updating to new standards as we go along.

File: demos/vulkaninfo.c [hunk 5]
```diff
@@ -426,6 +426,21 @@ static const char *vk_format_string(VkFormat fmt) {
     }
 }
 
+static const char *vk_present_mode_string(VkPresentModeKHR mode) {
```
Comment (mark-lunarg) [inline demos/vulkaninfo.c]: See comments below

File: demos/vulkaninfo.c [hunk 6]
```diff
@@ -426,6 +426,21 @@ static const char *vk_format_string(VkFormat fmt) {
...
+    switch (mode) {
+#define STR(r)                                                                 \
+    case VK_PRESENT_MODE_##r:                                                  \
+        return #r
+        STR(IMMEDIATE_KHR);
```
Comment (chrisforbes) [inline demos/vulkaninfo.c]: Can we just use `string_VkPresentModeKHR()` here?

**Pull request closed without merging**